tests/test_ui_variant.py

```
@@ -0,0 +1,145 @@
+    def test_no_user_attribute_on_request_returns_legacy(self):
+        request = self.factory.get("/?ui=revamp")
+        result = resolve_template(request, "home", "cms/index.html", "cms/index_revamp.html")
+        self.assertEqual(result, "cms/index.html")
+        self.assertEqual(request.ui_variant, "legacy")
+
+
+class UIVariantViewTests(TestCase):
+    def setUp(self):
+        self.staff_user = User.objects.create_user(
+            username="staffuser",
+            password="testpass",
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to argument: "password"

(S106)